src/data_structures.rs

```diff
@@ -40,13 +65,17 @@ pub trait PCCommitterKey:
     fn max_degree(&self) -> usize;
 
     /// Outputs the maximum degree supported by the committer key.
-    fn supported_degree(&self) -> usize;
+    fn segment_size(&self) -> usize;
 
     /// Returns the hash of `self` instance.
     fn get_hash(&self) -> &[u8];
 
+    /// Returns key length
+    fn get_key_len(&self) -> usize;
+
```

Comment:
Let us reduce the number of functions we serve here, to just having `fn max_degree()` and `fn get_hash()`, which simply return the maximum degree and the hash of the current key. I do not see any need to refer to the maximum degree of the parent key it was derived from.

Reply:
Done